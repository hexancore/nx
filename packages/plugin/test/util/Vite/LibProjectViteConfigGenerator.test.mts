import { LibProjectViteConfigGenerator } from '../../../src/util/Vite/index.mjs';
import { createFakeRuntimeHelper } from '../../helper/functions.js';

describe(LibProjectViteConfigGenerator.constructor.name, () => {

  const runtimeHelper = createFakeRuntimeHelper({
    globs: {},
    readFiles: {
      '/root/package.json': '{name: "@acme-work/source"}'
    }
  });
  const generator = new LibProjectViteConfigGenerator(runtimeHelper);

  test.each([{ mode: 'production' }, { mode: "development" }])('should generate with mode: $mode', ({ mode }) => {
    const current = generator.create({
      workspaceProjectDir: '/root/acme-work/libs/acme-some',
    })({ command: 'build', mode });

    current.plugins = [];

    expect(current).toMatchSnapshot();
  });
});