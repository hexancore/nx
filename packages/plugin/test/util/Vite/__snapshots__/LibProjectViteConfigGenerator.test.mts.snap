// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function should generate with mode: development 1`] = `
{
  "build": {
    "commonjsOptions": {
      "transformMixedEsModules": true,
    },
    "cssCodeSplit": true,
    "emptyOutDir": true,
    "lib": {
      "entry": "src/index.ts",
      "fileName": "index",
      "formats": [
        "es",
      ],
      "name": "acme-some",
    },
    "outDir": "../../dist/libs/acme-some",
    "reportCompressedSize": true,
    "rollupOptions": {
      "external": [
        "vue",
      ],
      "input": {},
      "output": {
        "assetFileNames": [Function],
        "chunkFileNames": [Function],
        "entryFileNames": [Function],
        "format": "esm",
        "globals": {
          "vue": "Vue",
        },
        "manualChunks": [Function],
        "paths": undefined,
        "preserveModules": false,
      },
    },
    "target": "modules",
  },
  "cacheDir": "../../tmp/vite/acme-some",
  "plugins": [],
  "root": "/root/acme-work/libs/acme-some",
  "test": {
    "coverage": {
      "provider": "v8",
      "reportsDirectory": "../../tmp/coverage/acme-some",
    },
    "environment": "jsdom",
    "globals": true,
    "include": [
      "test/**/*.test.{ts,mts,cts,tsx}",
    ],
    "reporters": [
      "default",
    ],
  },
}
`;

exports[`Function should generate with mode: production 1`] = `
{
  "build": {
    "commonjsOptions": {
      "transformMixedEsModules": true,
    },
    "cssCodeSplit": true,
    "emptyOutDir": true,
    "lib": {
      "entry": "src/index.ts",
      "fileName": "index",
      "formats": [
        "es",
      ],
      "name": "acme-some",
    },
    "outDir": "../../dist/libs/acme-some",
    "reportCompressedSize": true,
    "rollupOptions": {
      "external": [
        "vue",
      ],
      "input": {},
      "output": {
        "assetFileNames": [Function],
        "chunkFileNames": [Function],
        "entryFileNames": [Function],
        "format": "esm",
        "globals": {
          "vue": "Vue",
        },
        "manualChunks": [Function],
        "paths": undefined,
        "preserveModules": false,
      },
    },
    "target": "modules",
  },
  "cacheDir": "../../tmp/vite/acme-some",
  "plugins": [],
  "root": "/root/acme-work/libs/acme-some",
  "test": {
    "coverage": {
      "provider": "v8",
      "reportsDirectory": "../../tmp/coverage/acme-some",
    },
    "environment": "jsdom",
    "globals": true,
    "include": [
      "test/**/*.test.{ts,mts,cts,tsx}",
    ],
    "reporters": [
      "default",
    ],
  },
}
`;
